### ISSUE TITLE
Am I doing this right? (very slow)

### ISSUE THREAD

**Issue by ulatekh** [slowmoVideo/slowmoVideo#162]
I apologize for making this an "issue", but it was the only forum I could find.

I'm trying to use slowmoVideo to generate a 3x-slowdown version of an input clip. I don't need the variable-speed features of the GUI, so I'm using the command-line version. I'm building latest git under Fedora Core 29. My command is:

slowmoRenderer -target video "output.mkv" auto -input video "input.mkv" -slowfactor .35 -start :start -end :end -interpolation twoway2 -size orig

But after an hour, it doesn't seem to have produced anything. Looking at it in gdb, it appears to be in cv::FarnebackPrepareGaussian(), and never leaves. The lowest slowmoVideo-related stack frame is FlowSourceOpenCV_sV::buildFlowOpenCV_3(), called by FlowSourceOpenCV_sV::buildFlow() and Project_sV::requestFlow().

The input video is 448x256, 24000:1001 fps. My computer is a few years old at this point, but an 8-core 3.4GHz Intel Xeon should still be respectable. Other values for the -interpolation parameter behaved the same way.

Is it supposed to be this slow? Is my input video too large? Is there anything else I can do to track this down, except maybe building a debug version of opencv and digging around in cv::FarnebackPrepareGaussian()?

**Comment by valgit** [comment]
Hi @ulatekh ,
from what you said, I would bet foran GPU bug.
did you enable the GPU OpenCv code ? if so, could you disbale it for testing ?

**Comment by ulatekh** [comment]
I didn't do anything in particular to enable the GPU OpenCV code. From reading around (e.g. https://github.com/opencv/opencv/issues/9852), I can apparently disable OpenCL's runtime by setting environment variables (i.e. export OPENCV_OPENCL_RUNTIME=null and export OPENCV_OPENCL_DEVICE=disabled), and slowmoRenderer did output "Failed to load OpenCL runtime", but it's still stuck in cv::FarnebackPrepareGaussian().

**Comment by ulatekh** [comment]
I went through the trouble of building an OpenCV without GPU support (i.e. -D WITH_OPENCL=OFF and, out of paranoia, -DWITH_CUDA=OFF). Now it's stuck in cv::calcOpticalFlowFarneback(), which was 2 stack frames above cv::FarnebackPrepareGaussian() in the previous run.

**Comment by valgit** [comment]
@ulatekh when I said to disable the usage of GPU for OpenCV, it was in the preferences of the app. not : from memories, you can only do that on the GUI version but it's loaded by the CLI also.
can you try ?

**Comment by ulatekh** [comment]
I managed to get the GUI to (mostly) run to completion...although there were some other issues. The documentation (i.e. https://github.com/slowmoVideo/slowmoVideo/wiki/Documentation) doesn't actually say how to add a node (or what the difference between the arrow-icon and circle-icon in the upper-right is). Also, without a horizontal scroll bar, there's no way to create a slowed-down video that's outside of the aspect-ratio of one's screen. This is why I tried to use the command-line version.

Finally, selecting ffv1 as the video codec produces empty output, though I was able to recover by encoding the PNGs in the "rendered" directory (i.e. "ffmpeg -r 24000/1001 -i <path to rendered>/rendered-%05d.png -c:v ffv1 ~/Videos/rendered.mkv").

Don't get me wrong -- I appreciate your hard work, and your willingness to support your project. I'm just a bit frustrated with these issues.
